protocol/lavasession/common.go

```diff
@@ -112,6 +112,16 @@ func GetCaCertificate(serverCertPath, serverKeyPath string) (*tls.Config, error)
 	}, nil
 }
 
+func GetSelfSignedConfig() *tls.Config {
+	cert, err := GenerateSelfSignedCertificate()
+	if err != nil {
+		utils.LavaFormatFatal("failed to generate TLS certificate", err)
```

Comment:
this function shouldnt fatal, let's propagate the error outside the caller can decide on fatal or not